code/modules/antagonists/vampire/vamp_datum.dm

```diff
@@ -309,6 +309,8 @@
 
 /datum/antagonist/vampire/proc/vamp_burn(burn_chance)
 	if(prob(burn_chance) && owner.current.health >= 50)
+		if(!bloodtotal && !isspaceturf(get_turf(owner.current)))
+			return
```

Comment:
could you instead do this in `/datum/antagonist/vampire/proc/handle_vampire()`

you could just put an if check here to see if the vamp has a bloodtotal and not burn them if its 0, then you don't need to slowflake in a check like that
```dm
	if(istype(get_area(owner.current), /area/chapel) && !get_ability(/datum/vampire_passive/full))
		vamp_burn(7)
```